test/asm/foreach.asm

```diff
@@ -0,0 +1,21 @@
+foreach n, 10
+	printt "{d:n} "
+endr
+	printt "-> {d:n}\n"
+
+foreach x, 1, 5+1
+	printt "{d:x} "
+endr
+	printt "-> {d:x}\n"
+
+foreach v, 10, -1, -1
+	printt "{d:v} "
+v = 42
+endr
+	printt "-> {d:v}\n"
+
+s EQUS "x"
+foreach s, 3, 30, 3
+	printt "{d:x} "
+endr
+	printt "-> {d:x}\n"
```

Comment:
Test also for `PURGE x` within `FOREACH`

Reply:
What about `PURGE x ; x EQU 420` in the middle of a block as well? This should make it fail (can't re-assign at loop point)

Reply:
Added. This reports two errors, one for the failure to redefine `v` and one for the failure to update the `FOREACH` symbol which happens to be `v`.